cpp/src/slice2swift/Gen.cpp

```diff
@@ -1332,15 +1328,28 @@ Gen::ServantVisitor::visitOperation(const OperationPtr& op)
     for (const auto& param : op->inParameters())
     {
         const string typeString = typeToString(param->type(), op, param->optional());
-        out << param->mappedName() + ": " + typeString;
+        out << param->mappedName() + ": " + (param->type()->usesClasses() ? "sending " : "") + typeString;
     }
     out << ("current: " + getUnqualified("Ice.Current", swiftModule));
     out << epar;
 
     out << " async throws";
     if (op->returnsAnyValues())
     {
-        out << " -> " << operationReturnType(op);
+        out << " -> ";
+        auto params = op->returnAndOutParameters("");
```

Comment:
You should use instead Operation::returnsClasses. And it would be really nice to add some comments in Parser.h.